File: 040-049/ABC047_B.c
// ABC047 B - すぬけ君の塗り絵 2 イージー
#include <stdio.h>

int main(void)
{
    int W, H, N;

    scanf("%d %d %d", &W, &H, &N);

    int S[N][3];
    int i = 0;
    for (i = 0; i < N; i++) {
        scanf("%d %d %d", &S[i][0], &S[i][1], &S[i][2]);
    }

    int xStart = 0, xEnd = W, yStart = 0, yEnd = H;
    for (i = 0; i < N; i++) {
        switch (S[i][2]) {
        case 1:
            if (xStart < S[i][0]) {
                xStart = S[i][0];
            }
            break;
        case 2:
            if (xEnd > S[i][0]) {
                xEnd = S[i][0];
            }
            break;
        case 3:
            if (yStart < S[i][1]) {
                yStart = S[i][1];
            }
            break;
        case 4:
            if (yEnd > S[i][1]) {
                yEnd = S[i][1];
            }
            break;
        default:
            break;
        }
    }
    
    int result = (xEnd - xStart) * (yEnd - yStart);
    if (xEnd <= xStart || yEnd <= yStart) {
        result = 0;
    }

    printf("%d\n", result);
    return 0;
}

File: 040-049/ABC047_B.js
'use strict'

let rect = [];

function Main(input) {
    const d = input.split('\n').map(v => v.split(' '));
    let s = d.shift();
    const W = Number(s[0]);
    const H = Number(s[1]);
    const N = Number(s[2]);

    rect = [];
    for (let i = 0; i < H; i++) {
        for (let j = 0; j < W; j++) {
            rect[i] = Array(W).fill(0);
        }
    }
    for (let i = 0; i < N; i++) {
        switch (Number(d[i][2])) {
            case 1:
                draw(0, d[i][0], 0, H);
                break;
            case 2:
                draw(d[i][0], W, 0, H);
                break;
            case 3:
                draw(0, W, 0, d[i][1]);
                break;
            case 4:
                draw(0, W, d[i][1], H);
                break;
            default:
                break;
        }
    }

    let result = 0;
    for (const arr of rect) {
        for (const v of arr) {
            if (v === 0) {
                result++
            }
        }
    }

    console.log(result);
}

function draw(xStart, xEnd, yStart, yEnd) {
    for (let y = yStart; y < yEnd; y++) {
        for (let x = xStart; x < xEnd; x++) {
            rect[y][x] += 1;
        }
    }
}


Main(require('fs').readFileSync('/dev/stdin', 'utf8'));